week1/homework/src/server.js

```diff
@@ -9,12 +9,36 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    console.log(request.method, request.url);
+
+    response.writeHead(200, { 'Content-Type': 'application/json' });
+    switch (request.url) {
+      case '/state':
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/add':
+        state++;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/subtract':
+        state--;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/reset':
+        state = 10;
+        response.write(JSON.stringify({ state }));
+        break;
+      default:
+        const error = 'Not found';
+        response.writeHead(404, { 'Content-Type': 'application/json' });
```

Comment:
404 👍 

PRO TIP: you can also use `response.statusCode = 404;`

Reply:
```const error = 'Not found';
        response.writeHead(response.statusCode = 404, { 'Content-Type': 'application/json' });```
Do you mean this way?!

Reply:
I was thinking
```
response.statusCode = 404;
```
you can drop the `writeHead `line because  you already set the content type.
No big deal though

Reply:
```default:
        const error = 'Not found';
        response.statusCode = 404;
        response.write(JSON.stringify({ error }));
        break;```

This way?

Reply:
Yes that is what I meant. At the very start you have
`response.writeHead(200, { 'Content-Type': 'application/json' });`
so there is no need to specify the content type two times.

Reply:
But this way I got `1 test failed`
with this comment after :
`  Expected status code to be 404

  Difference:

  - 200
  + 404`
because the code status was defined as `200` at the very beginning.